pkg/controller/autoimport/autoimport_controller.go

```diff
@@ -69,109 +77,113 @@ func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, nil
 	}
 
-	autoImportSecret, err := r.informerHolder.AutoImportSecretLister.Secrets(managedClusterName).Get(constants.AutoImportSecretName)
-	if errors.IsNotFound(err) {
-		// the auto import secret could have been deleted, do nothing
+	if !managedCluster.DeletionTimestamp.IsZero() {
 		return reconcile.Result{}, nil
 	}
-	if err != nil {
-		return reconcile.Result{}, err
-	}
 
-	reqLogger.Info("Reconciling auto import secret")
+	if helpers.ClusterNotNeedToApplyImportResources(managedCluster) {
```

Comment:
Or we can separate the `ManagedClusterImportSucceeded` into two condition `ImportingResourcesApplied`(autoimport/selfmanagement/clusterdeployment) and `ManagedClusterImportSucceeded`(importstatus) ？